lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -338,6 +347,27 @@ def write(chunk)
       # Group the entries since we have to make one call per tag.
       grouped_entries = {}
       chunk.msgpack_each do |tag, *arr|
+        tag = tag.to_s
+        if @require_valid_tags
+          # When require_valid_tags is true, invalid cases will cause the log to
+          # be dropped and warnings generated in the log.
+          if convert_to_utf8(tag) != tag || tag.match(@invalid_tag_regexp)
+            @log.warn "Dropping log message(s) with invalid tag: '#{tag}'. " \
+                      'A tag should be a string with alphanumeric characters.'
+            next
+          end
+        else
+          # When require_valid_tags is false, try to convert any invalid tag by
+          # stripping off invalid characters.
+          converted_tag = convert_to_utf8(tag, '').gsub(@invalid_tag_regexp, '')
+          if converted_tag == ''
+            @log.warn "Dropping log message(s) with invalid tag: '#{tag}'. " \
```

Comment:
I've added the logic to return `'_'` when the original tag or the converted tag turned out to be `''`. Corresponding tests have been modified as well.